resdk/resolwe.py

```diff
@@ -111,6 +112,13 @@ def __init__(self, username=None, password=None, url=None):
 
         self.logger = logging.getLogger(__name__)
 
+    def login(self):
+        """Login to Resolwe server with username and password."""
+        username = input('Username: ')
+        password = getpass.getpass('Password: ')
+        self.auth = ResAuth(username, password, self.url)
```

Comment:
This code is duplicated.

Use this function also in `__init__`, you can change it to something like: `login(self, username=None, password=None, public=False)`.

Reply:
I would suggest `noninteractive=False` instead of `public=False`.

Reply:
I prefer `public`, because you can logout with `res.login(public=True)`. Or maybe you can add `logout` method for this. 

But i am against using private variables (i.e. `noninteractive`) in public functions. A solution would be to add third function that is called from both places.

Reply:
Why do you consider `noninteractive` as a "private variable"? It is just a way to enter or not enter credentials...

Reply:
To argue against public... What do you do if user enters public=True and also enters his username and password?